server/src/main/java/org/opensearch/index/search/QueryStringQueryParser.java

```diff
@@ -642,20 +670,24 @@ private Query getPossiblyAnalyzedPrefixQuery(String field, String termStr, Mappe
             Query posQuery;
             if (plist.size() == 1) {
                 if (isLastPos) {
+                    onDiscoveredField(currentFieldType.name());
                     posQuery = currentFieldType.prefixQuery(plist.get(0), getMultiTermRewriteMethod(), context);
                 } else {
+                    onDiscoveredField(field);
                     posQuery = newTermQuery(new Term(field, plist.get(0)), BoostAttribute.DEFAULT_BOOST);
                 }
             } else if (isLastPos == false) {
                 // build a synonym query for terms in the same position.
                 SynonymQuery.Builder sb = new SynonymQuery.Builder(field);
                 for (String synonym : plist) {
+                    onDiscoveredField(field);
```

Comment:
This does not look right: why we adding same field on each synonym? It should be done only once before

Reply:
done